### PR TITLE
OV-287: Add horizontal scroll in timeline

### LINKED ISSUE #287: BUG:  Timeline Overflow Issue - Missing Horizontal Scroll
### Describe your issue

When adding items to the horizontal timeline, there is no scroll functionality once the timeline becomes full. As a result, the "+" button disappears, preventing the user from adding more items.

### Steps to reproduce

1. Navigate to /studio, the timeline in the bottom of the page.
2. Begin 'adding scenes' to the timeline using the "+" button.
3. Continue adding items until the timeline is full.
4. Observe that there is no horizontal scroll bar or other means to scroll through the timeline.
5. Notice that the "+" button is no longer visible once the timeline is filled.

### What was the actual result?

Image

### What was the expected result?

When the timeline becomes full, a horizontal scroll bar should appear, or another method of navigating the timeline should be provided. The "+" button should always be visible, allowing the user to add more items.

### Put here any screenshots or videos (optional)

_No response_

### What OS are you seeing the problem on?

Windows

### What browsers are you seeing the problem on?

Chrome, Safari, Edge, Firefox, Opera

### Severity

Critical (S2)

### PR DESCRIPTION
**Pull request by XCODE89**
- Horizontal scroll was added to the timeline.
- Range readjustment was done when deleting or resizing a scene.

### CODE REVIEW

**Comment by XCODE89** [comment]
image

File: frontend/src/bundles/studio/components/timeline/timeaxis/timeaxis.tsx [hunk 1]
```diff
@@ -17,6 +17,8 @@ const TimeAxis: React.FC<Properties> = ({ markers }) => {
         useTimelineContext();
     const side = direction === 'rtl' ? 'right' : 'left';
 
+    const timelineWidth = range.end * 0.011;
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/studio/components/timeline/timeaxis/timeaxis.tsx]: fix magic value

File: frontend/src/bundles/studio/helpers/get-total-time.ts [hunk 2]
```diff
@@ -6,7 +6,10 @@ const calculateTotalMilliseconds = <T extends { duration: number }>(
 ): number => {
     const totalSeconds = items.reduce((sum, item) => sum + item.duration, 0);
     const totalMilliseconds = secondsToMilliseconds(totalSeconds);
-    return totalMilliseconds > rangeEnd ? totalMilliseconds : rangeEnd;
+    if (items.length > 4) {
+        return totalMilliseconds + 6000;
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/studio/helpers/get-total-time.ts]: same here

File: frontend/src/bundles/studio/helpers/get-total-time.ts [hunk 3]
```diff
@@ -6,7 +6,10 @@ const calculateTotalMilliseconds = <T extends { duration: number }>(
...
+    }
+    return totalMilliseconds >= rangeEnd ? totalMilliseconds + 6000 : rangeEnd;
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/studio/helpers/get-total-time.ts]: and here

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**